projects/driver_controls/test/test_pedal_calibration_test.c

```diff
@@ -0,0 +1,139 @@
+#include "ads1015_def.h"
+#include "debouncer.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "input_event.h"
+#include "interrupt.h"
+#include "log.h"
+#include "pedal_calibration.h"
+#include "pedal_calibration_fsm.h"
+#include "persist.h"
+#include "string.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+static PedalCalibrationStorage s_storage;
+static Ads1015Storage s_ads1015_storage;
+static ThrottleCalibrationData s_throttle_calibration_data;
+
+// Inputs user might want to change:
+#define TEST_PEDAL_CALIBRATION_ADC_CHANNEL_A ADS1015_CHANNEL_0
+#define TEST_PEDAL_CALIBRATION_ADC_CHANNEL_B ADS1015_CHANNEL_1
+#define TEST_PEDAL_CALIBRATION_BRAKE_PERCENTAGE 30
+#define TEST_PEDAL_CALIBRATION_COAST_PERCENTAGE 30
+#define TEST_PEDAL_CALIBRATION_TOELRANCE_SAFETY_FACTOR 2
+
+// The number of mocked readings for a state and a channel.
+#define TEST_PEDAL_CALIBRATION_NUM_MOCKED_READINGS 4
+
+// Counts every time both channels are read in a state.
+static uint8_t s_sample_counter[NUM_PEDAL_CALIBRATION_CHANNELS];
+
+// The array contains mocked readings for all channels and states.
+static int16_t s_mocked_reading[NUM_PEDAL_CALIBRATION_STATES][NUM_PEDAL_CALIBRATION_CHANNELS]
+                               [TEST_PEDAL_CALIBRATION_NUM_MOCKED_READINGS] = {
+                                 { { 600, 620, 630, 570 }, { 80, 120, 100, 90 } },
+                                 { { 1175, 1180, 1200, 1225 }, { 500, 470, 530, 510 } }
+                               };
+
+// The current state provided to the mocked ads1015_read_raw.
+static PedalCalibrationState s_current_state;
+
+// Mocks ads1015_read_raw. Reads from s_mocked_reading.
+StatusCode TEST_MOCK(ads1015_read_raw)(Ads1015Storage *storage, Ads1015Channel channel,
+                                       int16_t *reading) {
+  if (channel == TEST_PEDAL_CALIBRATION_ADC_CHANNEL_A) {
+    uint8_t nth_sample =
```

Comment:
Nit: turn this inner logic into a helper function. It is just repeated calls with different `PEDAL_CALIBRATION_CHANNEL_*`.

Reply:
You can pretty much ignore anything in this file - I haven't really looked at it and I'll probably rewrite it.